client/src/components/WorkoutSubmitForm/WorkoutSubmitForm.jsx

```diff
@@ -0,0 +1,272 @@
+  };
+
+  const removeUrl = (index) => {
+    const updatedUrls = [...existingFiles];
+    const removedItem = updatedUrls[index];
+    setFilesToRemove([...filesToRemove, removedItem]);
+    updatedUrls.splice(index, 1);
+    setExistingFiles(updatedUrls);
+  };
+
+  const handleFileChange = (files) => {
+    if (mediaFiles.length + existingFiles.length < 4) {
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        // Create a new object representing the media file with its URL
+        const newMediaFile = {
+          file: files[0],
+          url: reader.result,
+        };
+
+        // Update the mediaFiles array with the formatted URL
+        setMediaFiles([...mediaFiles, newMediaFile]);
+      };
+
+      // Read the contents of the selected image file as a data URL
+      reader.readAsDataURL(files[0]);
+    } else {
+      alert("You can only attach up to 4 files.");
+    }
+  };
+
+  const handleButtonClick = (type) => {
+    if (type === "image") {
+      imageInputRef.current.click();
+    } else if (type === "video") {
+      videoInputRef.current.click();
+    }
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    const formData = new FormData();
+
+    // Add the current datetime to the workoutInfo object
+    const currentDatetime = new Date().toISOString();
+    const updatedWorkoutInfo = {
+      ...workoutInfo,
+      existingMedia: existingFiles,
+      filesToRemove: filesToRemove,
+      dateTime: currentDatetime,
+    };
+
+    formData.append("json_data", JSON.stringify(updatedWorkoutInfo));
+    mediaFiles.forEach((file) => {
+      formData.append("files", file.file);
+    });
+
+    // Validate the workoutInfo object (customize the validation logic as needed)
+    // TODO perhaps add limits to length of name/description
+
+    // Submit the payload
+    const baseApiUrl = process.env.REACT_APP_API_URL;
+    const apiUrl =
+      mode === "create"
+        ? `${baseApiUrl}/api/insert-workout`
+        : `${baseApiUrl}/api/update-workout`;
+    try {
+      const token = sessionStorage.getItem("token");
+      const response = await fetch(apiUrl, {
+        method: mode === "create" ? "POST" : "PUT",
+        body: formData,
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      });
+
+      if (response.ok) {
+        setWorkoutInfo({
+          workoutName: "",
+          workoutDescription: "",
+        });
+        setMediaFiles([]);
+
+        if (mode === "edit") {
+          closeModal();
+        }
+
+        updateData();
+      } else {
+        throw new Error("Request failed");
+      }
+    } catch (error) {
+      console.error("Error uploading");
+      alert("Upload failed, please try again");
+    }
+  };
+
+  const handlePrivacyClick = () => {
+    console.log("Clicked on Privacy Dropdown");
+  };
+
+  return (
+    <>
+      <div className={workoutSubmitFormStyles["new-workout-container"]}>
+        <form
```

Comment:
If we want to further reduce the size of this component, this is where we'd start abstracting the <input> and <textarea> into its own component. 

Although the file is still kinda large (300+ lines), I think it's at its best balance without needing to make more smaller components